src/Psl/Str/trim.php

```diff
@@ -14,9 +16,13 @@
  * be stripped: space, tab, newline, carriage return, NUL byte, vertical tab.
  *
  * @pure
+ *
+ * @throws Psl\Exception\InvariantViolationException if $string is not a valid UTF-8 string.
  */
 function trim(string $string, ?string $char_mask = null): string
 {
+    Psl\invariant(is_utf8($string), 'Expected $string to be a valid UTF-8 string.');
```

Comment:
I like this approach since it validates the input.
@azjezz  mentioned using `Psl\Regex` would introduce some overhead. 
Pretty sure this `mb_detect_encoding` will cause some sort of overhead as well though, since the code will now validate twice for valid UTF-8 encoding: once in `is_utf8()` - and once in `preg_replace` because it contains the `u` modifier.

Should we benchmark this, or assume we're ok with this downside?